lang/cs/Org.Apache.REEF.IMRU/OnREEF/ResultHandler/WriteResultHandler.cs

```diff
@@ -0,0 +1,125 @@
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Remote.Impl;
+using Org.Apache.REEF.Wake.StreamingCodec;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.ResultHandler
+{
+    /// <summary>
+    /// Writes IMRU result from Update task to a file
+    ///  </summary>
+    /// <typeparam name="TResult"></typeparam>
+    public class WriteResultHandler<TResult> : IObserver<TResult>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof (WriteResultHandler<>));
+
+        private readonly IStreamingCodec<TResult> _resultCodec;
+        private readonly IFileSystem _fileSystem;
+        private readonly string _remoteFileName;
+        private readonly string _localFilename;
+
+        [Inject]
+        private WriteResultHandler(
+            IStreamingCodec<TResult> resultCodec,
+            IFileSystem fileSystem,
+            [Parameter(typeof(ResultOutputLocation))] string fileName)
+        {
+            _resultCodec = resultCodec;
+            _fileSystem = fileSystem;
+            _remoteFileName = fileName;
+            _localFilename = GenerateLocalFilename();
+        }
+
+        /// <summary>
+        /// Specifies how to handle the IMRU results from the Update Task. Writes it to a location
+        /// </summary>
+        /// <param name="value">The result of IMRU from the UpdateTask</param>
+        public void OnNext(TResult value)
+        {
+            if (string.IsNullOrWhiteSpace(_remoteFileName))
+            {
+                return;
+            }
+
+            WriteOutput(value);
+        }
+
+        /// <summary>
+        ///  Handles error scenario. Just throw it in this case
+        /// </summary>
+        /// <param name="error">Exception</param>
+        public void OnError(Exception error)
+        {
+            throw error;
+        }
+
+        /// <summary>
+        /// Handles what to do on completion
+        /// In this case write to remote location
+        /// </summary>
+        public void OnCompleted()
+        {
+            if (string.IsNullOrWhiteSpace(_remoteFileName))
+            {
+                return;
+            }
+
+            Uri remoteUri = _fileSystem.CreateUriForPath(_remoteFileName);
+
+            if (_fileSystem.Exists(remoteUri))
+            {
+                Exceptions.Throw(
+                    new Exception(string.Format("Output Uri: {0} already exists", remoteUri)), Logger);
+            }
+
+            _fileSystem.CopyFromLocal(_localFilename, remoteUri);
+        }
+
+        private string GenerateLocalFilename()
+        {
+            string localFileFolder = Path.GetTempPath() + "-partition-" + Guid.NewGuid().ToString("N").Substring(0, 8);
+            Directory.CreateDirectory(localFileFolder);
+            var localFilePath = String.Format("{0}\\{1}", localFileFolder, Guid.NewGuid().ToString("N").Substring(0, 8));
+
+            if (File.Exists(localFilePath))
```

Comment:
unless when the output file path is passed to the task, the folder is already exist. But there is a possibility that the directory doesn't exist.